app/controllers/github_controller.rb

```diff
@@ -213,12 +213,27 @@ def pullapprove_merge_hook
         return
       end
 
+      unless Dir.exist?('SmokeDetector')
+        `git clone https://github.com/Charcoal-SE/SmokeDetector`
+
+        File.write 'SmokeDetector/.git/info/attributes', <<~END
+          bad_keywords.txt -text merge=union
+          blacklisted_usernames.txt -text merge=union
+          blacklisted_websites.txt -text merge=union
+          watched_keywords.txt -text merge=union
+        END
+      end
+
       if !Octokit.client.pull_merged?('Charcoal-SE/SmokeDetector', pr_num)
-        Octokit.client.merge_pull_request(
-          'Charcoal-SE/SmokeDetector',
-          pr_num,
-          '--autopull'
-        )
+        Dir.chdir('SmokeDetector') do
+          ref = pr[:head][:ref]
+
+          `git checkout master; git pull origin master`
+          `git fetch origin #{ref}`
+          `git merge origin/#{ref} --no-ff -m 'Merge pull request ##{pr_num} from Charcoal-SE/#{ref} --autopull'`
+          `git push origin master`
```

Comment:
Is there an Octokit equivalent for each of these commands?

Reply:
No, sadly. The Github API does not respect the `merge=union` Git attribute (otherwise I could just drop a `.gitattributes` in the SmokeDetector repo).

Reply:
https://github.com/isaacs/github/issues/487